packages/network-connector/src/index.ts

```diff
@@ -1,47 +1,90 @@
 import { ConnectorUpdate } from '@web3-react/types'
 import { AbstractConnector } from '@web3-react/abstract-connector'
-import Web3ProviderEngine from 'web3-provider-engine'
-import CacheSubprovider from 'web3-provider-engine/subproviders/cache.js'
-import { RPCSubprovider } from '@0x/subproviders/lib/src/subproviders/rpc_subprovider' // https://github.com/0xProject/0x-monorepo/issues/1400
 import invariant from 'tiny-invariant'
 
 interface NetworkConnectorArguments {
   urls: { [chainId: number]: string }
   defaultChainId?: number
-  pollingInterval?: number
-  requestTimeoutMs?: number
+}
+
+// taken from ethers.js, compatible interface with web3 provider
+type AsyncSendable = {
+  isMetaMask?: boolean;
+  host?: string;
+  path?: string;
+  sendAsync?: (request: any, callback: (error: any, response: any) => void) => void;
+  send?: (request: any, callback: (error: any, response: any) => void) => void;
+}
+
+class MiniRpcProvider implements AsyncSendable {
+  public readonly isMetaMask: false = false
+  public readonly chainId: number
+  public readonly url: string
+  public readonly host: string
+  public readonly path: string
+
+  constructor(chainId: number, url: string) {
+    this.chainId = chainId
+    this.url = url
+    const parsed = new URL(url)
+    this.host = parsed.host
+    this.path = parsed.pathname
+  }
+
+  sendAsync(request: any, callback: (error: any, response: any) => void): void {
+    fetch(this.url, {
+      method: 'POST',
+      body: JSON.stringify(request)
+    })
+      .then(
+        result => {
+          return result.json()
+            .then(
+              json => {
+                if (!result.ok) {
+                  throw new Error(`${json.code}: ${json.message}`)
+                }
+                return json
+              }
+            )
+        }
+      )
+      .then(
+        json => {
+          callback(undefined, json)
+        }
+      )
+      .catch(
+        error => {
+          callback(error, undefined)
+        }
+      )
+  }
```

Comment:
Consider extracting out most of this into an async/await function, adding support for EIP 1193's `request` (future compatible) and improving error handling and checking:
```ts
class MiniRpcProvider {
	public readonly url: string
	public readonly sendAsync = (request: { jsonrpc: '2.0', id: number | string | null, method: string, params?: unknown[] | object}, callback: (error: any, response: any) => void): void => {
		this.request(request.method, request.params)
			.then(result => callback(null, { jsonrpc: '2.0', id: request.id, result }))
			.catch(error => callback(error, null))
	}

	public readonly request = async (method: string, params?: unknown[] | object): Promise<unknown> => {
		const response = await fetch(this.url, {
			method: 'POST',
			body: JSON.stringify({
				jsonrpc: '2.0',
				id: 1,
				method,
				params,
			}),
		})
		if (!response.ok) throw new RequestError(`${response.status}: ${response.statusText}`, -32000)
		const body = await response.json()
		if ('error' in body) {
			// TODO: validate that body.error.message and body.error.code exist
			throw new RequestError(body.error.message, body.error.code, body.error.data)
		} else if ('result' in body) {
			return body.result
		} else {
			throw new RequestError(`Received unexpected JSON-RPC response to ${method} request.`, -32000, body)
		}
	}
}

class RequestError extends Error {
	constructor(message: string, public code: number, public data?: unknown) {
		super(message)
	}
}
```

Reply:
Should validate whether `sendAsync` is expecting the error in its callback to be a JS `Error` or a JSON-RPC error object.  This code assumes a JS `Error`, but if it is expecting a JSON-RPC error then you'll need to convert.